components/expand-collapse/expand-collapse.js

```diff
@@ -0,0 +1,144 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+const states = {
+	PRECOLLAPSING: 'precollapsing', // setting up the styles so the collapse transition will run
+	COLLAPSING: 'collapsing', // in the process of collapsing
+	COLLAPSED: 'collapsed', // fully collapsed
+	PREEXPANDING: 'preexpanding', // setting up the styles so the expand transition will run
+	EXPANDING: 'expanding', // in the process of expanding
+	EXPANDED: 'expanded', // fully expanded
+};
```

Comment:
Question: what happens if someone does this in their initial markup?

```html
<d2l-expand-collapse expanded>Stuff in here</d2l-expand-collapse>
```

Does it animate right away? I think ideally we'd only animate user-initiated state changes. If they want the thing expanded initially, chances are they "saved" the state and are just keeping it in that same state -- like we do with widgets on the homepage.

Reply:
That's a good call, I think I should be able to make that work so there is no animation. I'll give look into it. Right now it works but if the page loads quickly you do see a bit of a transition.

Reply:
I'm trying to think of a scenario where we'd want the animation initially and I'm coming up empty. It's always going to look weird and jarring and it'll force a bunch of repaints and cause other content below it to move and reflow.

Reply:
Yeah my current thinking is there should be no animation initially.